### PR TITLE
NAS-115365 / 22.12 / NAS-115365: Refactoring DatasetTrivialPermissionsComponent

### PR DESCRIPTION
**Pull request by undsoft**
Create a dataset and go to View Permissions -> Edit.

### CODE REVIEW

File: src/app/pages/storage/volumes/permissions/containers/dataset-trivial-permissions/dataset-trivial-permissions.component.ts [hunk 1]
```diff
@@ -1,323 +1,205 @@
...
-    }
-    if (!data.apply_group) {
-      delete data.group;
-    }
-    delete data.apply_user;
-    delete data.apply_group;
+  private preparePayload(): DatasetPermissionsUpdate {
+    const values = this.form.value;
 
-    data['acl'] = [];
```
Comment (AlexKarpov98) [inline src/app/pages/storage/volumes/permissions/containers/dataset-trivial-permissions/dataset-trivial-permissions.component.ts:263]: There is a missing - `acl` is set empty array.

**Review by AlexKarpov98 — CHANGES_REQUESTED** [review]
Screenshot_6

File: src/app/pages/storage/volumes/permissions/containers/dataset-trivial-permissions/dataset-trivial-permissions.component.ts [hunk 2]
```diff
@@ -1,323 +1,205 @@
...
-          title: this.translate.instant('Warning'),
-          message: this.translate.instant('Setting permissions recursively will affect this directory and any others below it. This might make data inaccessible.'),
-        }).pipe(
-          filter((confirmed) => !confirmed),
-          untilDestroyed(this),
-        ).subscribe(() => {
-          recursive.setValue(false);
-        });
-      }
+    const dialogRef = this.matDialog.open(EntityJobComponent, {
```
[the remark this answers is not in the document]
Comment (undsoft) [reply src/app/pages/storage/volumes/permissions/containers/dataset-trivial-permissions/dataset-trivial-permissions.component.ts:118]: I deliberately didn't choose to change this. I think EntityJobComponent is used to show progress when there are a lot of files that need to be updated when recursive flag is used.

**Review by AlexKarpov98 — APPROVED** [review]
LGTM! 👍

**Pull request merged**